### ISSUE TITLE
Issues in training multiple models separately but simultaneously in Pytorch

### ISSUE THREAD

**Issue by boyuzz** [horovod/horovod#921]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.1.0a0+f36f3cc
3. Horovod version: 0.15.2
4. MPI version: 4.0.0
5. CUDA version: 9.2.88
6. NCCL version: 2.4.2
7. Python version: 3.6.8
8. OS and version: Red Hat 4.8.5-36

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.md)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.md)?

**Your question:**
I want to use horovod to train multiple models **separately but simultaneously**, where each model is trained on one GPU (Each node has 2 GPUs in my platform). In the [pytorch example](https://github.com/horovod/horovod/blob/master/examples/pytorch_mnist.py), can I do that by using a train_loader without the distributed sampler and the original pytorch optimizer (not the horovod distributed optimizer)?

**Comment by tgaddair** [comment]
Hey @boyuzz, yes, you can train multiple models simultaneously with (or without) Horovod.

If you only want to train one model per GPU, you can throw out most of the Horovod components for data parallelism, gradient aggregation, learning rate adjustment, etc.  The only thing you need to do is ensure that each process is pinned to a separate gpu:

```
hvd.init()
torch.cuda.set_device(hvd.local_rank())
```

Then you need to use some conditional logic to determine which model you're going to train:

```
if hvd.rank() == 0:
    model = create_model1()
elif hvd.rank() == 1:
    model = create_model2()
```

The rest is normal PyTorch.  The application is still driven using `mpirun` as usual.

A more interesting use case would be training multiple models across multiple GPUs at once.  For example, if you want to train multiple models with multiple GPUs per model.  For that, you'd want to look into the comm param in [hvd.init()](https://github.com/horovod/horovod/blob/master/horovod/common/__init__.py#L62).

**Comment by boyuzz** [comment]
@tgaddair Thank you for your answer. The last point you mentioned sounds interesting. But are there any examples about that? E.g., training 5 models with 10 GPUs where each model is trained on 2 GPUs? I can't find any examples until now. Thank you.

**Comment by alsrgv** [comment]
@boyuzz, can you clarify why you want to train models simultaneously?  Would all models have exactly the same training duration?

**Comment by boyuzz** [comment]
@alsrgv Yes, all models have the same training duration. I want to transfer some information among models during training. That's why I want to train them simultaneously.

**Comment by alsrgv** [comment]
@boyuzz, you could do this... You could use `mpi4py` to share "some information", and use `hvd.init(comm=[list of ranks])` (https://github.com/horovod/horovod/blob/master/horovod/common/__init__.py#L58) to partition your set of workers into distinct Horovod rings.
